dspace-api/src/main/java/org/dspace/harvest/dao/impl/HarvestedCollectionDAOImpl.java

```diff
@@ -29,108 +30,143 @@
+            status));
+        return list(context, criteriaQuery, false, HarvestedCollection.class, -1, -1);
     }
 
     @Override
     public HarvestedCollection findByCollection(Context context, Collection collection) throws SQLException {
-        Criteria criteria = createCriteria(context, HarvestedCollection.class);
-        criteria.add(Restrictions.eq("collection", collection));
-        return singleResult(criteria);
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, HarvestedCollection.class);
+        Root<HarvestedCollection> harvestedCollectionRoot = criteriaQuery.from(HarvestedCollection.class);
+        criteriaQuery.select(harvestedCollectionRoot);
+        criteriaQuery.where(criteriaBuilder.equal(harvestedCollectionRoot.get(HarvestedCollection_.collection),
+            collection));
+        return singleResult(context, criteriaQuery);
 
     }
 
     @Override
-    public List<HarvestedCollection>
-        findByLastHarvestedAndHarvestTypeAndHarvestStatusesAndHarvestTime(Context context, Date startTime,
-                                                                          int minimalType, int[] statuses,
-                                                                          int expirationStatus, Date expirationTime)
-        throws SQLException {
-//      Old query: "SELECT * FROM harvested_collection WHERE
-// (last_harvested < ? or last_harvested is null) and harvest_type > ? and (harvest_status = ? or harvest_status = ?
-// or (harvest_status=? and harvest_start_time < ?)) ORDER BY last_harvested",
-//                new java.sql.Timestamp(startTime.getTime()), 0, HarvestedCollection.STATUS_READY,
-// HarvestedCollection.STATUS_OAI_ERROR, HarvestedCollection.STATUS_BUSY, new java.sql.Timestamp(expirationTime
-// .getTime()));
-        Criteria criteria = createCriteria(context, HarvestedCollection.class);
-        LogicalExpression lastHarvestedRestriction = Restrictions.or(
-            Restrictions.lt("lastHarvested", startTime),
-            Restrictions.isNull("lastHarvested")
+    public List<HarvestedCollection> findByLastHarvestedAndHarvestTypeAndHarvestStatusesAndHarvestTime(Context context,
+        Date startTime, int minimalType, int[] statuses, int expirationStatus,
+        Date expirationTime) throws SQLException {
+
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, HarvestedCollection.class);
+        Root<HarvestedCollection> harvestedCollectionRoot = criteriaQuery.from(HarvestedCollection.class);
+        criteriaQuery.select(harvestedCollectionRoot);
+
+        Predicate orPredicate = criteriaBuilder.or(criteriaBuilder.lessThan(harvestedCollectionRoot.get
+                (HarvestedCollection_.lastHarvested), startTime),
```

Comment:
Does this need to be treated as a <Date>?  I noticed that elsewhere in this PR, and this also looks like a Date.

Reply:
Other examples of this in this same class.